src/main/scala/io/iohk/ethereum/faucet/jsonrpc/FaucetBuilder.scala

```diff
@@ -0,0 +1,112 @@
+package io.iohk.ethereum.faucet.jsonrpc
+
+import java.security.SecureRandom
+
+import akka.actor.ActorSystem
+import io.iohk.ethereum.faucet.FaucetConfigBuilder
+import io.iohk.ethereum.jsonrpc.server.controllers.ApisBase
+import io.iohk.ethereum.jsonrpc.server.controllers.JsonRpcBaseController.JsonRpcConfig
+import io.iohk.ethereum.jsonrpc.server.http.JsonRpcHttpServer
+import io.iohk.ethereum.keystore.KeyStoreImpl
+import io.iohk.ethereum.mallet.service.RpcClient
+import io.iohk.ethereum.utils.{ConfigUtils, KeyStoreConfig, Logger}
+
+import scala.util.Try
+
+trait ActorSystemBuilder {
+  def systemName: String
+  implicit lazy val system: ActorSystem = ActorSystem(systemName)
+}
+
+trait FaucetControllerBuilder {
+  self: FaucetConfigBuilder with ActorSystemBuilder =>
+
+  implicit val ec = system.dispatcher
+}
+
+trait FaucetRpcServiceBuilder {
+  self: FaucetConfigBuilder with FaucetControllerBuilder with ActorSystemBuilder =>
+
+  val keyStore = new KeyStoreImpl(KeyStoreConfig.customKeyStoreConfig(faucetConfig.keyStoreDir), new SecureRandom())
+  val rpcClient = new RpcClient(faucetConfig.rpcAddress)
+  val faucetRpcService = new FaucetRpcService(rpcClient, keyStore, faucetConfig)
+}
+
+trait FaucetJsonRpcHealthCheckBuilder {
+  self: FaucetRpcServiceBuilder =>
+
+  val faucetJsonRpcHealthCheck = new FaucetJsonRpcHealthCheck(faucetRpcService)
```

Comment:
IMHO it would be easier to test/reason about if `faucetJsonRpcHealthCheck` would be passed as an argument to class rather than summon somewhere deep in the inheritance hierarchy (traits mixin).